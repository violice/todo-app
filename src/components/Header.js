import React from 'react';
import { createUseStyles } from 'react-jss'

const useStyles = createUseStyles({
  headerContainer: {
    backgroundColor: '#673ab7',
    boxShadow: '0px 2px 4px -1px rgba(0,0,0,0.2), 0px 4px 5px 0px rgba(0,0,0,0.14), 0px 1px 10px 0px rgba(0,0,0,0.12)',
    color: '#fff',
    marginBottom: '64px',
  },
  headerContent: {
    maxWidth: '1200px',
    height: '64px',
    margin: '0 auto',
    padding: '0px 24px',
    display: 'flex',
    alignItems: 'center',
  },
  logo: {
    fontSize: '24px',
    fontWeight: 500,
  },
});

const Header = () => {
  const styles = useStyles();
  return (
    <header className={styles.headerContainer}>
      <div className={styles.headerContent}>
        <span className={styles.logo}>TODO</span>
      </div>
    </header>
  )
};

export default Header;