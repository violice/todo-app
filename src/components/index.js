export { default as AuthForm } from './AuthForm';
export { default as TaskForm } from './TaskForm';
export { default as Header } from './Header';
export { default as TasksList } from './TasksList';