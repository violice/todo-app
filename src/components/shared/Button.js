import React from 'react';
import PropTypes from 'prop-types';
import { createUseStyles } from 'react-jss';

const useStyles = createUseStyles({
  button: {
    backgroundColor: '#673ab7',
    color: '#fff',
    border: 'none',
    borderRadius: '4px',
    padding: '8px 16px',
    minWidth: '96px',
    '&:focus': {
      outline: 'none',
    },
    '&:hover': {
      backgroundColor: '#8561c5',
      cursor: 'pointer',
    },
  },
});

const Button = ({ title, onClick }) => {
  const styles = useStyles();
  return (
    <button className={styles.button} onClick={onClick}>{title}</button>
  );
};

Button.propTypes = {
  title: PropTypes.string.isRequired,
  onClick: PropTypes.func,
};

Button.defaultProps = {
  onClick: () => {},
};

export default Button;