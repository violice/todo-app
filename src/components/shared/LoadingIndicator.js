import React from 'react';
import PropTypes from 'prop-types';
import { createUseStyles } from 'react-jss';

const useStyles = createUseStyles({
  overlay: {
    position: 'absolute',
    zIndex: '100',
    height: '100%',
    width: '100%',
    top: '0',
    left: '0',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    background: 'rgba(255,255,255,.75)',
    borderRadius: '8px',
  },
  loading: {
    margin: '0 auto',
    display: 'block',
    position: 'relative',
    width: '40px',
    height: '40px',
    '&>div': {
      boxSizing: 'border-box',
      display: 'block',
      position: 'absolute',
      width: '32px',
      height: '32px',
      margin: '4px',
      border: '4px solid #673ab7',
      borderRadius: '50%',
      animation: 'loading-animation 1.2s cubic-bezier(0.5, 0, 0.5, 1) infinite',
      borderColor: '#673ab7 transparent transparent transparent',
      '&:nth-child(1)': {
        animationDelay: '-0.45s',
      },
      '&:nth-child(2)': {
        animationDelay: '-0.3s',
      },
      '&:nth-child(3)': {
        animationDelay: '-0.15s',
      },
    },
  },
});

const LoadingIndicator = ({ withOverlay }) => {
  const styles = useStyles();

  if (withOverlay) {
    return (
      <div className={`${styles.overlay}`}>
        <div className={styles.loading}>
          <div></div>
          <div></div>
          <div></div>
          <div></div>
        </div>
      </div>
    );
  }

  return (
    <div className={styles.loading}>
      <div></div>
      <div></div>
      <div></div>
      <div></div>
    </div>
  );
};

LoadingIndicator.propTypes = {
  withOverlay: PropTypes.bool,
};

LoadingIndicator.defaultProps = {
  withOverlay: false,
};

export default LoadingIndicator;