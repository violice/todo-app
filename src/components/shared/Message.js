import React from 'react';
import PropTypes from 'prop-types';
import { createUseStyles } from 'react-jss';

const useStyles = createUseStyles({
  message: {
    backgroundColor: '#FFE4E9',
    borderRadius: '4px',
    color: '#FF1A43',
    fontSize: '12px',
    padding: '8px 16px',
  },
});

const Message = ({ value }) => {
  const styles = useStyles();
  return (
    <span className={styles.message}>{value}</span>
  );
};

Message.propTypes = {
  value: PropTypes.string.isRequired,
};

export default Message;