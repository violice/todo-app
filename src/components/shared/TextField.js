import React from 'react';
import PropTypes from 'prop-types';
import { createUseStyles } from 'react-jss';

const useStyles = createUseStyles({
  textFieldContainer: {
    marginBottom: '16px',
    borderBottom: '1px solid rgba(0,0,0,.87)',
    padding: '4px',
  },
  textFieldLabel: {
    display: 'block',
    marginBottom: '8px',
    color: 'rgba(0,0,0,.54)',
    fontSize: '12px',
  },
  textField: {
    display: 'block',
    width: '100%',
    padding: '0px',
    border: 'none',
    backgroundColor: 'transparent',
    '&:focus': {
      outline: 'none',
    },
  },
});

const TextField = ({ title, value, multiline, onChange }) => {
  const styles = useStyles();
  return (
    <div className={styles.textFieldContainer}>
      <label className={styles.textFieldLabel}>{title}</label>
      {multiline
        ? <textarea
          className={styles.textField}
          name="username"
          value={value}
          onChange={e => onChange(e.target.value)}
        />
        : <input
          className={styles.textField}
          name="username"
          value={value}
          onChange={e => onChange(e.target.value)}
        />
      }
    </div>
  );
};

TextField.propTypes = {
  title: PropTypes.string,
  value: PropTypes.string.isRequired,
  multiline: PropTypes.bool,
  onChange: PropTypes.func.isRequired,
};

TextField.defaultProps = {
  title: '',
  multiline: false,
};

export default TextField;