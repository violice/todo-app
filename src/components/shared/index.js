export { default as Button } from './Button';
export { default as LoadingIndicator } from './LoadingIndicator';
export { default as Message } from './Message';
export { default as TextField } from './TextField';