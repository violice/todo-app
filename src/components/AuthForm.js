import React from 'react';
import PropTypes from 'prop-types';
import { createUseStyles } from 'react-jss';
import { Button, TextField, Message, LoadingIndicator } from 'components/shared';

const useStyles = createUseStyles({
  container: {
    maxWidth: '320px',
    padding: '32px',
    margin: '0 auto',
    backgroundColor: '#fff',
    boxShadow: '0 2px 1px -1px rgba(0,0,0,.2), 0 1px 1px 0 rgba(0,0,0,.14), 0 1px 3px 0 rgba(0,0,0,.12)',
    borderRadius: '8px',
    position: 'relative',
  },
  messageContainer: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'center',
  },
  buttonContainer: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'center',
    marginTop: '32px',
  },
});

const AuthForm = ({ username, password, isLoading, error, onUsernameChange, onPasswordChange, onFormSubmit }) => {
  const styles = useStyles();

  const onSubmit = (e) => {
    e.preventDefault();
    onFormSubmit({ username, password });
  }

  return (
    <form className={styles.container} onSubmit={onSubmit}>
      {isLoading && <LoadingIndicator withOverlay />}
      <TextField title="Username" value={username} onChange={onUsernameChange} />
      <TextField title="Password" value={password} onChange={onPasswordChange} />
      {error &&
        <div className={styles.messageContainer}>
          <Message value={error.message} />
        </div>
      }
      <div className={styles.buttonContainer}>
        <Button title="SignIn / SignUp" />
      </div>
    </form>
  );
};

AuthForm.propTypes = {
  username: PropTypes.string.isRequired,
  password: PropTypes.string.isRequired,
  isLoading: PropTypes.bool,
  error: PropTypes.object,
  onUsernameChange: PropTypes.func.isRequired,
  onPasswordChange: PropTypes.func.isRequired,
  onFormSubmit: PropTypes.func.isRequired,
};

AuthForm.defaultProps = {
  isLoading: false,
  error: null,
};

export default AuthForm;