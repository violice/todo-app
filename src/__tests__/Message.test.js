import React from 'react';
import { render } from '@testing-library/react';
import { Message } from 'components/shared';

it('should take a snapshot', () => {
  const { asFragment } = render(<Message value="test message" />)
  expect(asFragment()).toMatchSnapshot();
});