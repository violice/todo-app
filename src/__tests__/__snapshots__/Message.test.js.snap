// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <span
    class="message-0-2-1"
  >
    test message
  </span>
</DocumentFragment>
`;
